pkg/apis/core/validation/validation.go

```diff
@@ -4217,10 +4201,7 @@ func ValidateServiceUpdate(service, oldService *core.Service) field.ErrorList {
 		if oldService.Spec.Type != core.ServiceTypeExternalName && oldService.Spec.ClusterIP != "" {
 			allErrs = append(allErrs, ValidateImmutableField(service.Spec.ClusterIP, oldService.Spec.ClusterIP, field.NewPath("spec", "clusterIP"))...)
 		}
-		// notes:
-		// we drop the IPFamily field when the Dualstack gate is off.
-		// once the gate is on, we start assigning default ipfamily according to cluster settings. in other words
-		// though the field is immutable, we allow (onetime) change from nil==> to value
+		// a user is not allowed to change the IPFamily field, but they are allowed to nil it out (old clients that don't know about IPFamily)
```

Comment:
this is misleading and doesn't match the check below. from validation's perspective, they're not allowed to nil it out... PrepareForUpdate preserves the old value

Reply:
will update